### ISSUE TITLE
Fix Request: false positives and missed cases in UnresolvedType

### ISSUE THREAD

**Issue by jborgers** [jborgers/PMD-jPinpoint-rules#544]
False positives for lombok @ Slf4j en method references op @ Data en @ Value objecten en interfaces.

Missed case for static import, #469

**Comment by jborgers** [comment]
Examples

```
@Service
@Slf4j
public class MyProducerBad1 {

    public void delete(String idStr) {
        log.debug("delete --> {}", idStr); // bad: no import 
    }
}
```

```
import lombok.extern.slf4j.Slf4j;
@Service
public class MyProducerBad2 {

    public void delete(String idStr) {
        log.debug("delete --> {}", idStr); // bad: no @Slf4j
    }
}
```

```
import lombok.extern.slf4j.Slf4j;
@Service
@Slf4j
public class MyProducerGood {

    public void delete(String idStr) {
        log.debug("delete --> {}", idStr); // good:  import and @Slf4j
    }
}
```

**Comment by stokpop** [comment]
```
    @Bean
    public ConfigurableServletWebServerFactory webServerFactory() {
        TomcatServletWebServerFactory factory = new TomcatServletWebServerFactory();
        factory.addConnectorCustomizers(connector -> connector.setProperty("prop", "test")); // <- false positive
        return factory;
    }
```

**Comment by jborgers** [comment]
This seems a shortcoming of PMD. When you provide a cast, it is not a violation:
`factory.addConnectorCustomizers((Connector)connector -> connector.setProperty("prop", "test"));`

**Comment by jborgers** [comment]
Simplified test case with provided dependencies:

```
class Issue544_lambda {
    public TomcatServletWebServerFactory webServerFactory() {
        TomcatServletWebServerFactory factory = new TomcatServletWebServerFactory();
        factory.addConnectorCustomizers(connector -> connector.setProperty("prop", "test")); // <- false positive on 'connector'
        return factory;
    }
}

class TomcatServletWebServerFactory {
    void addConnectionCustomizers(TomcatConnectorCustomizer... connector) {}
}

@FunctionalInterface
public interface TomcatConnectorCustomizer {
    void customize(Connector connector);
}
public class Connector {
    public void setProperty(String name, String val) {}
}
```
